sklearn/model_selection/_validation.py

```diff
@@ -637,6 +654,65 @@ def _multimetric_score(estimator, X_test, y_test, scorers):
     return scores
 
 
+def _apply_scorer(estimator, X, y, scorer, sample_weight):
+    """Applies the scorer to the estimator, given the data and sample_weight.
+
+    If ``sample_weight`` is None or contains all ones, ``sample_weight`` WILL
+    NOT be passed to ``scorer``; otherwise, it will be passed.
+
+    In the event that ``sample_weight`` is provided and used but ``scorer``
+    doesn't accept a ``sample_weight`` parameter, then a ``TypeError`` should
+    likely be raised.
+
+    Parameters
+    ----------
+    estimator : estimator object implementing 'fit'
+        The object that was used to fit the data.
+
+    X : array-like of shape at least 2D
+        The data to fit.
+
+    y : array-like
+        The target variable to try to predict in the case of
+        supervised learning.  (May be None)
+
+    scorer : A single callable.
+        Should return a single float.
+
+        The callable object / fn should have signature
+        ``scorer(estimator, X, y, sample_weight=None)`` if ``sample_weight``.
+
+    sample_weight : array-like, shape (y)
+        sample weights to use during metric calculation.  May be None.
+
+    Returns
+    -------
+    score : float
+        Score returned by ``scorer`` applied to ``X`` and ``y`` given
+        ``sample_weight``.
+    """
+    if sample_weight is None or np.all(sample_weight == 1):
+        if y is None:
+            score = scorer(estimator, X)
+        else:
+            score = scorer(estimator, X, y)
+    else:
+        try:
```

Comment:
Is there any benefit to passing `sample_weight` as a keyword argument?  E.g.

```python
kwargs = { 'sample_weight': sample_weight}

# ...

score = score(estimator, X, y, **kwargs)
```

The reason I ask is that when I found the other PR that did importance weighting (https://github.com/scikit-learn/scikit-learn/pull/10806/files#diff-60033c11a662f460e1567effd5faa6f0R584) they had this.  I don't know if that's important or not.  

What does everyone think?